services/puppetforge/puppetforge-modules.service.js

```diff
@@ -1,281 +0,0 @@
-      },
-    ]
-  }
-
-  static registerLegacyRouteHandler() {}
-}
-
-class PuppetforgeModuleEndorsement extends LegacyService {
-  static get category() {
-    return 'rating'
-  }
-
-  static get route() {
-    return {
-      base: 'puppetforge/e',
-      pattern: ':user/:moduleName',
-    }
-  }
-
-  static get examples() {
-    return [
-      {
-        title: 'Puppet Forge endorsement',
-        namedParams: {
-          user: 'camptocamp',
-          moduleName: 'openssl',
-        },
-        staticPreview: {
-          label: 'endorsement',
-          message: 'approved',
-          color: 'green',
-        },
-      },
-    ]
-  }
-
-  static registerLegacyRouteHandler() {}
-}
-
-class PuppetforgeModuleFeedback extends LegacyService {
-  static get category() {
-    return 'rating'
-  }
-
-  static get route() {
-    return {
-      base: 'puppetforge/f',
-      pattern: ':user/:moduleName',
-    }
-  }
-
-  static get examples() {
-    return [
-      {
-        title: 'Puppet Forge feedback score',
-        namedParams: {
-          user: 'camptocamp',
-          moduleName: 'openssl',
-        },
-        staticPreview: {
-          label: 'score',
-          message: '61%',
-          color: 'yellow',
-        },
-      },
-    ]
-  }
-
-  static registerLegacyRouteHandler() {}
-}
-
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-class PuppetforgeModules extends LegacyService {
-  static get category() {
-    return 'other'
-  }
-
-  static get route() {
-    return {
-      base: 'puppetforge',
-      pattern: '([^/]+)/([^/]+)/([^/]+)',
-    }
-  }
-
-  static registerLegacyRouteHandler({ camp, cache }) {
-    camp.route(
-      /^\/puppetforge\/([^/]+)\/([^/]+)\/([^/]+)\.(svg|png|gif|jpg|json)$/,
-      cache((data, match, sendBadge, request) => {
-        const info = match[1] // either `v`, `dt`, `e`, `f`, or `p`
-        const user = match[2]
-        const module = match[3]
-        const format = match[4]
-        const options = {
-          json: true,
-          uri: `https://forgeapi.puppetlabs.com/v3/modules/${user}-${module}`,
-        }
-        const badgeData = getBadgeData('puppetforge', data)
-        request(options, (err, res, json) => {
-          if (err != null || (json.length !== undefined && json.length === 0)) {
-            badgeData.text[1] = 'inaccessible'
-            sendBadge(format, badgeData)
-            return
-          }
-          try {
-            if (info === 'v') {
-              if (json.current_release) {
-                const version = json.current_release.version
-                badgeData.text[1] = versionText(version)
-                badgeData.colorscheme = versionColor(version)
-              } else {
-                badgeData.text[1] = 'none'
-                badgeData.colorscheme = 'lightgrey'
```

Comment:
I've just ignored this case. If we hit a response that doesn't have the `current_release` key, we'll fail to validate with "unparseable json repsonse". I think this is probably reasonable.

Reply:
👍